### PR TITLE
added assets and rates by lowercase contract address

### PR DESCRIPTION
**Pull request by poocart**
Adds asset rates and asset balances by lowercase asset contract address.

### CODE REVIEW

File: src/utils/common.js [hunk 1]
```diff
@@ -595,3 +595,8 @@ export const formatBigFiatAmount = (value: Value, fiatCurrency: string) => {
 export const getEnsName = (username: string) => `${username}${getEnsPrefix()}`;
 
 export const extractUsernameFromEnsName = (ensName: string) => ensName.replace(getEnsPrefix(), '');
+
+export const valueAsKey = (value: string | number): string => {
```
Comment (mdjastrzebski) [inline src/utils/common.js:599]:
```suggestion
export const addressAsKey = (value: string): string => {
```

Other potential names to consider `toAddressKey`, `keyFromAddress`, etc

Naming should reflect that this is intended for address keys, I think that we do not want to assume that each record value needs to be lowercase. wdyt?

File: src/utils/common.js [hunk 2]
```diff
@@ -595,3 +595,8 @@ export const formatBigFiatAmount = (value: Value, fiatCurrency: string) => {
...
+  if (!value) return '';
+  return value.toString().toLowerCase();
+};
```
Comment (mdjastrzebski) [inline src/utils/common.js:602]: I would also add 
```
export const valueForAddress = <Value>(record: Record<Value>, address: ?string): ?Value => {
   return record[valueAsKey(address)];
}
```

Which would slightly improve call site by reducing parenthesis nesting & correctly indicate that the key might return nil value.

```
if (!balances[valueAsKey(gasTokenAddress)]) return false;
```
 =>
```
if (!valueForAddress(balances, gasTokenAddress)) return false;
```

File: src/utils/rates.js [hunk 3]
```diff
@@ -36,7 +36,7 @@ export const getUsdToFiatRate = (rates: RatesByAssetAddress, currency: Currency)
   if (currency === USD) return 1;
 
   // will select native asset rates to calculate between
-  const nativeAssetRates = rates[ethers.constants.AddressZero];
+  const nativeAssetRates = rates[valueAsKey(ethers.constants.AddressZero)];
```
Comment (mdjastrzebski) [inline src/utils/rates.js:39]: Can we have zero address re-exported from e.g. assetConstants. Importing ethers every time is a bit tedious.

File: src/components/ValueInput/ValueInput.js [hunk 4]
```diff
@@ -168,7 +169,7 @@ const ValueInputComponent = ({
   const assetSymbol = assetData?.symbol || '';
   const chain = assetData?.chain || CHAIN.ETHEREUM;
   const walletBalances = accountAssetsBalances?.[chain]?.wallet ?? {};
-  const assetBalance = (customBalances || walletBalances)?.[assetAddress]?.balance || '0';
+  const assetBalance = (customBalances || walletBalances)?.[valueAsKey(assetAddress)]?.balance || '0';
```
Comment (mdjastrzebski) [inline src/components/ValueInput/ValueInput.js:172]:
```suggestion
  const assetBalance = (customBalances ?? walletBalances)?.[valueAsKey(assetAddress)]?.balance || '0';
```

**Review by mdjastrzebski — APPROVED** [review]
Looks great!

Suggested some minor improvements that could help with readability.

File: src/utils/common.js (hunk 1 above)
Comment (poocart) [reply src/utils/common.js:599]: Honestly I would prefer this for more generic name, but we use it for address only as for now so I guess it's ok, replaced

**Pull request merged**